concurrent/src/main/java/com/facebook/airlift/concurrent/ConcurrentScheduledExecutor.java

```diff
@@ -0,0 +1,166 @@
+package com.facebook.airlift.concurrent;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ConcurrentScheduledExecutor
+        implements ScheduledExecutorService
+{
+    private final List<ScheduledExecutorService> schedulers;
+
+    public ConcurrentScheduledExecutor(int schedulerCount, int threadsPerScheduler, String threadNameFormat)
+    {
+        checkArgument(schedulerCount > 0, "schedulerCount must be at least one");
+        checkArgument(threadsPerScheduler > 0, "threadsPerScheduler must be at least one");
+        ThreadFactory threadFactory = daemonThreadsNamed(requireNonNull(threadNameFormat, "threadBaseName is null"));
+
+        ImmutableList.Builder<ScheduledExecutorService> schedulersBuilder = ImmutableList.builder();
+        for (int i = 0; i < schedulerCount; i++) {
+            ScheduledThreadPoolExecutor scheduledExecutorService = new ScheduledThreadPoolExecutor(threadsPerScheduler, threadFactory);
+            scheduledExecutorService.setRemoveOnCancelPolicy(true);
```

Comment:
Please extract it as a parameter. Enabling it by default might not be intuitive.